ecospheres_universe/feed_universe.py

```diff
@@ -137,43 +122,39 @@ def get_organization_datasets(self, org: str):
                                       or d.get('private') or d.get('extras'))}
         return self.get_datasets(url, func, xfields=xfields)
 
-    def get_topic_datasets(self, topic, organization_id: str = ""):
-        url = f"{self.base_url}/api/2/topics/{topic}/datasets/?page_size=1000"
-        if organization_id:
-            url += f"&organization={organization_id}"
-        func = lambda c: {d['id'] for d in c['data']}
-        return self.get_datasets(url, func)
-
-    def get_topic_datasets_v1(self, topic):
-        url = f"{self.base_url}/api/1/topics/{topic}/"
-        xfields = 'datasets{id}'
-        func = lambda c: {d['id'] for d in c['datasets']}
-        return self.get_datasets(url, func, xfields=xfields)
-
-    def search_datasets_count(self, topic_id: str | None, organization_id: str = ""):
-        if not topic_id:
-            return
-        url = f"{self.base_url}/api/2/datasets/search/?topic={topic_id}"
-        if organization_id:
-            url += f"&organization={organization_id}"
+    def get_topic_datasets_count(self, topic_id: str, org_id: str, use_es: bool = False):
+        url = f"{self.base_url}/api/2/datasets{'/search' if use_es else ''}/?topic={topic_id}&organization={org_id}&page_size=1"
         r = session.get(url)
         r.raise_for_status()
-        return r.json()['total']
+        return r.json()["total"]
+
+    def get_topic_datasets_elements(self, topic: str) -> list[tuple[str, str]]:
+        elements = []
+        url = f"{self.base_url}/api/2/topics/{topic}/elements/?class=Dataset&page_size=1000"
+        while True:
+            r = session.get(url)
+            r.raise_for_status()
+            c = r.json()
+            elements.extend([(elt['id'], elt['element']['id']) for elt in c['data']])
```

Comment:
Would read nicer when used with a `@dataclass` instead of an anonymous tuple.

Reply:
Went with a NamedTuple, but yeah.